packages/player/src/components/video.tsx

```diff
@@ -51,35 +59,48 @@ export const Video: React.FC<VideoProps> = ({
   //to the videoService and update our state machine context
   const videoElemRef = React.useRef<any>(null)
   const videoService = useVideo()
+  const paused = useSelector(videoService, selectPaused)
+  const readyState = useSelector(videoService, selectReadyState)
+  const currentTime = useSelector(videoService, selectCurrentTime)
+  const autoplay = useSelector(videoService, selectAutoplay)
 
   useMetadataTrackList()
 
+  React.useEffect(() => {
+    videoService.send({type: 'REGISTER', videoRef: videoElemRef})
+
+    // sometimes the event handlers aren't registered before the
+    // `canPlay` event is fired (caching, for instance) so we want
+    // to check and see if it's ready as soon as we get a ref to
+    // the HTMLVideoElement and "manually" fire an event
+
+    videoElemRef.current.readyState > 3 && videoService.send('LOADED')
+  }, [videoElemRef])
+
   return (
     <video
       className={cx([`cueplayer-react-video`, className])}
       id={id}
       crossOrigin={crossOrigin}
       ref={(c: HTMLVideoElement) => {
         videoElemRef.current = c
-        videoService.send({type: 'REGISTER', videoRef: videoElemRef})
-
-        // sometimes the event handlers aren't registered before the
-        // `canPlay` event is fired (caching, for instance) so we want
-        // to check and see if it's ready as soon as we get a ref to
-        // the HTMLVideoElement and "manually" fire an event
-        if (c && c.readyState > 3) {
-          videoService.send('LOADED')
-        }
       }}
       muted={muted}
       preload={preload}
       loop={loop}
       playsInline={playsInline}
-      autoPlay={autoPlay}
+      autoPlay={false}
       poster={poster}
       src={src}
+      onClickCapture={(_event) => {
+        videoService.send(paused ? 'PLAY' : 'PAUSE')
+      }}
       onCanPlay={(_event) => {
         videoService.send('LOADED')
+        // only autoplay if player is ready, paused, and at the beginning
+        if (autoplay && paused && readyState > 3 && currentTime === 0) {
+          videoService.send('PLAY')
+        }
```

Comment:
this combination seems to be working perfectly

Reply:
this is asking for issues and we should stop implementing this style of autoplay and "autoplay a playlist" in the player library